dask_jobqueue/core.py

```diff
@@ -290,7 +298,8 @@ def job_file(self):
             yield fn
 
     def _submit_job(self, script_filename):
-        return self._call(shlex.split(self.submit_command) + [script_filename])
+        return self._call(shlex.split(self.submit_command) + [script_filename],
+                          cwd=self.log_directory)
```

Comment:
Hmmm, this may have unintended side-effects, although I am not 100% sure ... for example if I specify some relative path (e.g. for data files) in my notebook they may not be found in the dask-worker process.

Is it not possible to pass a folder to `-e` and `-o` (SGE allows this but maybe this is very specific to SGE ...)?

I guess if I understand https://github.com/dask/dask-jobqueue/issues/141#issuecomment-417225928, there is no way to have variable expansion in PBS directive. I googled a bit and found http://community.pbspro.org/t/pbs-jobid-variable/176 which does not seem to leave too much hope ... I have seen "solutions" that used `qalter` after the submission e.g. [this](https://stackoverflow.com/a/28186725) maybe that's an option but that seems quite complicated ...

Just curious, so what do people do with PBSPro? They don't set the stdout/stderr file, they remember to change it each time with a different name?

Reply:
Yes you are right, this may not be the best approach...

You've understood the problem with PBS, using -e or -o is possible but quite limited. Default output file name is $JOB_NAME.o$JOB_ID. So usually I just leave this and don't set anything.

Another solution is to just specify a folder. In this case PBS will fill it with $JOB_ID.OUT files.

Reply:
> Another solution is to just specify a folder. In this case PBS will fill it with $JOB_ID.OUT files.

If a specifying a folder is possible I think I would just do that for PBS (and SGE as well).